benchmarks/bench_isotonic.py

```diff
@@ -31,9 +31,15 @@ def generate_logistic_dataset(size):
     return np.random.random(size=size) < 1.0 / (1.0 + np.exp(-X))
 
 
+def generate_pathological_dataset(size):
+    # Triggers O(n^2) complexity on the original implementation.
+    return np.r_[np.arange(n), np.arange(-(n - 1), n), np.arange(-(n - 1), 1)]
```

Comment:
It probably caught the global `n` (with the same value) set in the `if __name__ == "__main__"` block...